### ISSUE TITLE
Support custom network/subnet for Google Cloud Life Sciences API executor

### ISSUE THREAD

**Issue by snamburi3** [nextflow-io/nextflow#1574]
## New feature

Ability to specify a custom network and subnet for the analysis. Presently nextflow uses default network in the Google cloud project(GCP). 

## Usage scenario 
The GCP project comes with global network. We disable and create a custom network in the project for compliance reasons and avoid accidentlal egress costs between regions.

## Suggest implementation 
It will great if nextflow executor could support specifying [custom network and subnet](https://cloud.google.com/life-sciences/docs/reference/rest/v2beta/projects.locations.pipelines/run#Network)

**Comment by Simsulation** [comment]
+1 

@moschetti - same issue we talked about

**Comment by moschetti** [comment]
This would be similar to the implementation of private addresses in [this commit](https://github.com/nextflow-io/nextflow/commit/a481b42b152ad002eab01f8df8823e877d0e8e5b), but would instead add `Network.network` & `Network.subnetwork`

**Comment by slagelwa** [comment]
A much needed ability +1

**Comment by hnawar** [comment]
+1
I can try out a solution and submit a PR

**Comment by hnawar** [comment]
To give more context, this is particularly needed where organizations may need to change the Default network if they are connecting to on-prem using Interconnect through a Shared VPC.
I think an implementation similar to the CPU platform in the Life Sciences Profile would be the best option

**Comment by slagelwa** [comment]
+1

**Comment by galvinma** [comment]
+1

I am attempting to run a workflow, but cannot find a shared VPC:

 ```
Execution failed: creating instance: inserting instance: Invalid value for field 'resource.networkInterfaces[0].network': ''. The referenced network resource cannot be found.
```

**Comment by pditommaso** [comment]
Solved. Credits @hnawar

**Comment by pditommaso** [comment]
Re-opening since still not working https://github.com/nextflow-io/nextflow/pull/1810#issuecomment-740604470

**Comment by pditommaso** [comment]
Solved by 171f4aa8
